browser/conftest.py

```diff
@@ -40,3 +42,87 @@ def _db(app):
     """
     db = SQLAlchemy(app=app)
     return db
+
+# File fixtures
+HEIGHT = 32
+WIDTH = 32
+RES = (HEIGHT, WIDTH)
+FRAMES = 5
+CHANNELS = 3
+FEATURES = 2
+
+TEST_NAMES = ["empty", "full1", "full2", "iden", "tril1",
+              "triu1", "tril2", "triu2", "triu1l2", "tril1u2",
+              "checkerboard01", "checkerboard12"]
+tri_12 = np.triu(np.ones(RES, dtype=np.int16))
+tri_12[tri_12 == 0] = 2
+tri_21 = np.tril(np.ones(RES, dtype=np.int16))
+tri_21[tri_21 == 0] = 2
+check01 = np.array([[0, 1], [1, 0]], dtype=np.int16)
+check12 = np.array([[1, 2], [2, 1]], dtype=np.int16)
+TEST_FRAMES = [np.zeros(RES, dtype=np.int16),  # empty
+               np.ones(RES, dtype=np.int16),  # all ones
+               np.full(RES, 2, dtype=np.int16),  # all twos
+               np.identity(HEIGHT, dtype=np.int16),  # identity
+               np.tril(np.ones(RES, dtype=np.int16)),  # lower triangular ones
+               np.triu(np.ones(RES, dtype=np.int16)),  # upper triangular ones
+               np.tril(np.full(RES, 2, dtype=np.int16)),  # lower triangular twos
+               np.triu(np.full(RES, 2, dtype=np.int16)),  # upper triangular twos
+               tri_12,
+               tri_21,
+               np.tile(check01, (HEIGHT // 2, WIDTH // 2)),
+               np.tile(check12, (HEIGHT // 2, WIDTH // 2)),
+               ]
+assert len(TEST_NAMES) == len(TEST_FRAMES)
+
+
+def repeat_feature(feature, n):
+    """Repeats a feature n times along the last axis."""
+    return np.repeat(
+        np.expand_dims(feature, axis=-1),
+        n,
+        axis=-1)
+
+
+def repeat_frame(frame, n):
+    """Repeats a frame n times along the first axis."""
+    return np.repeat(
+        np.expand_dims(frame, axis=0),
+        n,
+        axis=0)
+
+
+@pytest.fixture(params=TEST_FRAMES, ids=TEST_NAMES)
+def zstack_file(mocker, request):
+    def load(self):
+        data = {'raw': np.zeros((FRAMES, HEIGHT, WIDTH, CHANNELS))}
+        data['annotated'] = repeat_frame(repeat_feature(request.param, FEATURES), FRAMES)
+        return data
+    mocker.patch('files.CalibanFile.load', load)
+    return CalibanFile('filename.npz', 'bucket', 'path')
```

Comment:
It might be useful to try testing a 2D image (`rgb` mode).

Reply:
I added a single frame test image, which I think is what we're looking for. This exercises the dimension expansion code (starting with `if self.raw.ndim == 3:`) in the CalibanFile `__init__` that was previously grouped with the RGB logic.